src/test/java/io/numaproj/numaflow/function/FunctionServerTestErr.java

```diff
@@ -0,0 +1,202 @@
+                .setValue(inValue)
+                .build();
+
+        var stub = UserDefinedFunctionGrpc.newBlockingStub(inProcessChannel);
+        try {
+            stub.mapTFn(inDatum);
+        } catch (Exception e) {
+            assertEquals(Status.UNKNOWN.getCode().toString(), e.getMessage());
+        }
+    }
+
+    @Test
+    public void reducerWithOneKey() {
+        String reduceKey = "reduce-key";
+
+        Metadata metadata = new Metadata();
+        metadata.put(Metadata.Key.of(WIN_START_KEY, Metadata.ASCII_STRING_MARSHALLER), "60000");
+        metadata.put(Metadata.Key.of(WIN_END_KEY, Metadata.ASCII_STRING_MARSHALLER), "120000");
+
+        //create an output stream observer
+        ReduceOutputStreamObserver outputStreamObserver = new ReduceOutputStreamObserver();
+
+        try {
+            StreamObserver<Udfunction.DatumRequest> inputStreamObserver = UserDefinedFunctionGrpc
+                    .newStub(inProcessChannel)
+                    .withInterceptors(MetadataUtils.newAttachHeadersInterceptor(metadata))
+                    .reduceFn(outputStreamObserver);
+
+            for (int i = 1; i <= 10; i++) {
+                Udfunction.DatumRequest inputDatum = Udfunction.DatumRequest.newBuilder()
+                        .setValue(ByteString.copyFromUtf8(String.valueOf(i)))
+                        .addKeys(reduceKey)
+                        .build();
+                inputStreamObserver.onNext(inputDatum);
+            }
+
+            inputStreamObserver.onCompleted();
+
+        } catch (Exception e) {
+            assertEquals(Status.UNKNOWN.getCode().toString(), e.getMessage());
+        }
+    }
+
+    @Test
+    public void reducerWithMultipleKey() {
+        String reduceKey = "reduce-key";
+
+        Metadata metadata = new Metadata();
+        metadata.put(Metadata.Key.of(WIN_START_KEY, Metadata.ASCII_STRING_MARSHALLER), "60000");
+        metadata.put(Metadata.Key.of(WIN_END_KEY, Metadata.ASCII_STRING_MARSHALLER), "120000");
+
+        //create an output stream observer
+        ReduceOutputStreamObserver outputStreamObserver = new ReduceOutputStreamObserver();
+
+        try {
+            StreamObserver<Udfunction.DatumRequest> inputStreamObserver = UserDefinedFunctionGrpc
+                    .newStub(inProcessChannel)
+                    .withInterceptors(MetadataUtils.newAttachHeadersInterceptor(metadata))
+                    .reduceFn(outputStreamObserver);
+            for (int i = 1; i <= 10; i++) {
+                Udfunction.DatumRequest inputDatum = Udfunction.DatumRequest.newBuilder()
+                        .setValue(ByteString.copyFromUtf8(String.valueOf(i)))
+                        .addKeys(reduceKey)
+                        .build();
+                inputStreamObserver.onNext(inputDatum);
+            }
+
+            inputStreamObserver.onCompleted();
+        } catch (Exception e) {
+            assertEquals(Status.UNKNOWN.getCode().toString(), e.getMessage());
+        }
+    }
```

Comment:
is this part even getting executed?

Reply:
yes, it assert if the status code is `unknown`.

Reply:
is the catch block getting executed?

Reply:
yes